firebase-perf/src/test/java/com/google/firebase/perf/internal/AppStateMonitorTest.java

```diff
@@ -387,6 +388,7 @@ public void screenTrace_perfMonDisabledAtBuildTime_traceNotCreated() {
     Activity activityWithNonHardwareAcceleratedView =
         createFakeActivity(/* isHardwareAccelerated =*/ true);
     ConfigResolver configResolver = ConfigResolver.getInstance();
+    configResolver.setDeviceCacheManager(new DeviceCacheManager(new FakeDirectExecutorService()));
```

Comment:
We should not need to set Device Cache in any other test that doesn't directly talk to Device Cache. Is any test erroring out if we don't set this?

*Same for all references below.

Reply:
Since the getSharedPreferences call is now kicked off in a background thread, we need to make sure that tests uses the FakeDirectExecutorService. Otherwise, there might be a race condition with when the getSharedPreferences are available if it is down in a background thread.

Calls like `setIsPerformanceCollectionEnabled` use the device manager under the hood and can have different results depending on when getSharedPreferences is enabled.